homeassistant/components/smartthings/fan.py

```diff
@@ -70,14 +102,32 @@ async def _async_set_percentage(self, percentage: int | None) -> None:
         # the entity state ahead of receiving the confirming push updates
         self.async_write_ha_state()
 
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set the preset_mode of the fan."""
+        if (FanEntityFeature.PRESET_MODE in self._attr_supported_features) and (
+            self.preset_modes is not None
+        ):
+            if preset_mode in self.preset_modes:
+                await self._device.set_fan_mode(preset_mode, set_status=True)
+
+            self.async_write_ha_state()
+
     async def async_turn_on(
         self,
         percentage: int | None = None,
         preset_mode: str | None = None,
         **kwargs: Any,
     ) -> None:
         """Turn the fan on."""
-        await self._async_set_percentage(percentage)
+        if FanEntityFeature.SET_SPEED in self._attr_supported_features:
+            # If speed is set in features then turn the fan on with the speed.
+            await self._async_set_percentage(percentage)
```

Comment:
Whatif `percentage` is `None`?

Reply:
It is checked in https://github.com/home-assistant/core/blob/dev/homeassistant/components/smartthings/fan.py#L62